### ISSUE TITLE
Still active?

### ISSUE THREAD

**Issue by poperigby** [jeaye/q3#59]
Hi. Is this still active, @jeaye? It looks like a cool project.

**Comment by jeaye** [comment]
@PopeRigby Nope! It's been about 6 years since I was focusing on it at all. Rust was still very young, back then, and changing often. Keeping up was tiresome. It's a lovely language, though, and I'm glad it's seeing the success it is.

As for q3, I may revisit it in the future, but I'm in [startup mode](https://okletsplay.com) these days, so I don't have much time for side projects.

Thanks for your interest! Rust has some awesome stuff going on in its gamedev community; be sure to check it out.

**Comment by poperigby** [comment]
Yeah. There are some super cool games made in Rust. I might try forking this and continuing it's development once I learn some Rust.

**Comment by jeaye** [comment]
Have at it! q3 is under the BSD-3 license. Since the Rust is 6 years old, there will be a *lot* of catching up first. :)

**Comment by poperigby** [comment]
BSD-3 is GPL compatible, correct?

**Comment by jeaye** [comment]
I'm not a lawyer. This link claims it is, however: https://directory.fsf.org/wiki/License:BSD-3-Clause

More info is also here: https://www.gnu.org/licenses/license-list.html#GPLCompatibleLicenses

**Comment by poperigby** [comment]
Seems like it should be good. Why did you choose the BSD license, if you don't mind me asking?

**Comment by jeaye** [comment]
I chose the BSD over the GPL because I felt more comfortable understanding all of its terms without being a lawyer.

These days, for most things, I use my own license: https://upload.jeaye.com/jank-license

It's a variant of the Sleepycat license, which is GPL-compatible and strictly copyleft, in contrast with the BSD 3-Clause. I've switched to this because it's the simplest GPL-compatible copyleft license I've found. Well, Sleepycat was, and then I removed the portion about needing to pay $$ to get the source to arrive at the license I use. Now the license I use is the simplest GPL-compatible copyleft license of which I know.

**Comment by poperigby** [comment]
I see. So is your license just a simplified GPL? Is there a reason you haven't switched this repo to that license?

**Comment by jeaye** [comment]
Yeah, pretty much a simplified GPL. No real reason q3 hasn't been switched over. For most of my other projects, there are many other contributors, so it's a hassle to switch licenses. For q3, I could easily, since I'm the only contributor, but I haven't since I don't think it matters much in this case. The benefits of copyleft over copyright for q3 would imply I'm concerned someone will fork q3, close source it, and then do nothing but give me some attribution. There are two reasons why that isn't a concern:

1. q3 isn't actively developed, so I could push a commit now, bringing on my new license, but everything prior to that commit is still licensed BSD 3-Clause. i.e. the whole repo

2. The time it would take someone to catch q3 up to the latest Rust, with the intention of closing the source and most likely selling the product, would not be worth starting from q3, probably; this task could be much more simply done in Unity or UE4 or Godot, so it's simply not practical that someone would want to do it

With that said, if I were to start active development on q3 again, it's fair to say I might update the license to apply to the newer versions.
